pkg/referrer/referrer.go

```diff
@@ -54,7 +59,7 @@ func (r *referrer) checkReferrer(ctx context.Context, ref string, manifestDigest
 
 		// Parse image manifest list from referrers.
 		var index ocispec.Index
-		bytes, err := io.ReadAll(rc)
+		bytes, err := io.ReadAll(io.LimitReader(rc, 0x100000))
```

Comment:
It seems we can't suppose the manifest index size.

Reply:
I would prefer a reasonable limit here, otherwise it may be an attacker point for DoS.